### PR TITLE
pybind/mgr: check for empty metadata mgr_module:get_metadata()

### PR DESCRIPTION
**Pull request by vshankar**
... rather than just None - looks like we could get back an
empty dictionary when the metadata isn't populated (which is
asynchronous).

Fixes: http://tracker.ceph.com/issues/57072
Signed-off-by: Venky Shankar <vshankar@redhat.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Review by kotreshhr — APPROVED** [review]

File: src/pybind/mgr/mgr_module.py [hunk 1]
```diff
@@ -1564,7 +1564,7 @@ def get_metadata(self,
         :rtype: dict, or None if no metadata found
         """
         metadata = self._ceph_get_metadata(svc_type, svc_id)
-        if metadata is None:
+        if not metadata:
```
Comment (ajarr) [inline src/pybind/mgr/mgr_module.py:1567]: The doc string says `None` is returned if no metadata is available. That should be changed? Or is this a bug in _ceph_get_metadata() that it doesn't return None for 'mds' service but does so for others when no metadata is available?

Comment (vshankar) [reply src/pybind/mgr/mgr_module.py:1567]: The metadata (map) could be empty IMO.

Comment (ajarr) [reply src/pybind/mgr/mgr_module.py:1567]: I think I misunderstood. Ignore my comment.

**Review by neesingh-rh — APPROVED** [review]
It looks like metadata is empty dict. at that time which wasn't handled, for which assertion error was thrown, from logs. The fix looks good.

File: src/pybind/mgr/mgr_module.py (hunk 1 above)
Comment (ajarr) [reply src/pybind/mgr/mgr_module.py:1567]: So we want defaultdict(lambda: 'unknown') to be returned when _ceph_get_metadata() returns empty dictionary for mds service? okay

**Review by joscollin — APPROVED** [review]

**Comment by rishabh-d-dave** [comment]
jenkins test api

**Comment by rishabh-d-dave** [comment]
jenkins test api

**Review by rishabh-d-dave — APPROVED** [review]
https://tracker.ceph.com/projects/cephfs/wiki/Main#2022-Sep-29

**Pull request merged**